### PR TITLE
Less tolerant of messy builds

### LINKED ISSUE #235: infer ensimeScalaVersion from majority vote
currently we just use the root's `scalaVersion` but many projects are setup incorrectly to use sbt's scala version for the root. @dwijnand

### LINKED ISSUE #268: create missing directories with option to skip
nobody is paying attention to the warning

### LINKED ISSUE #269: exception if there are sources in root
nobody is paying attention to the error. Require an override to acknowledge

### PR DESCRIPTION
**Pull request by fommil**
- close #268 
- close #269 
- close #235

### CODE REVIEW

**Comment by fommil** [comment]
@dwijnand @sjrd now that we are using coursier to download the scala repl and ensime server, I was able to implement a majority vote algorithm to determine the version of scala to use. Mixed scala projects require the user to provide an override to acknowledge that ensime doesn't support multiple scala versions, all instructions are spat out in the sbt exception. Because people read the instructions... (lol, rotfl...)

File: src/main/scala/EnsimePlugin.scala [hunk 1]
```diff
@@ -29,6 +30,16 @@ object EnsimeKeys {
...
   )
 
+  val ensimeIgnoreSourcesInBase = settingKey[Boolean](
+    "ENSIME doesn't support sources in base, this tolerates their existence."
+  )
+  val ensimeIgnoreMissingDirectories = settingKey[Boolean](
+    "ENSIME requires declared source directories to exist. If you find this noisy, ignore them."
+  )
+  val ensimeIgnoreScalaMismatch = settingKey[Boolean](
+    "ENSIME can only use one version of scala per project. If you have mixed scala versions (e.g. lib and sbt plugin), set this to acknowledge."
```
Comment (dwijnand) [inline src/main/scala/EnsimePlugin.scala:40]: "per project" is misleading. Maybe go for something like "for the entire build" or "for the build definition".

File: src/main/scala/EnsimePlugin.scala [hunk 2]
```diff
@@ -370,34 +384,31 @@ object EnsimePlugin extends AutoPlugin {
...
            |  scalaVersion in ThisBuild := "${scalaVersion.gimme}"
-           |in your build.sbt or add
+           |in your build.sbt, or override the ENSIME scala version with:
            |  ensimeScalaVersion in ThisBuild := "${scalaVersion.gimme}"
-           |to a local ensime.sbt: http://ensime.org/build_tools/sbt/#customise""".stripMargin
+           |in a ensime.sbt: http://ensime.org/build_tools/sbt/#customise""".stripMargin
       )
+      if (!ensimeIgnoreScalaMismatch.gimme)
+        throw new IllegalStateException(
+          """To ignore this error (i.e. have have multiple scala versions), customise `ensimeIgnoreScalaMismatch`"""
```
Comment (dwijnand) [inline src/main/scala/EnsimePlugin.scala:397]: typo "have have"

Comment (fommil) [reply src/main/scala/EnsimePlugin.scala:397]: argh. any chance you could just make these typo fixups and send a quick PR? they sound good and you can do it directly in github

Comment (dwijnand) [reply src/main/scala/EnsimePlugin.scala:397]: I feel like I'm entitled to force you to do all the work, damn it. jk ;-) #282

Comment (fommil) [reply src/main/scala/EnsimePlugin.scala:397]: :stuck_out_tongue:

**Pull request merged**